src/lib/bcmp/bcmp.cpp

```diff
@@ -131,6 +158,23 @@ int32_t bmcp_process_packet(struct pbuf *pbuf, ip_addr_t *src, ip_addr_t *dst) {
       break;
     }
 
+    // Check if this type is one that will use sequence numbers
+    if (_message_is_sequenced_reply(header->type) && !_message_is_sequenced_request(header->type)) {
+      // Check if the message is a reply to a message we sent
+      bcmp_element_t *sent_message = _message_list_find_message(header->seq_num);
+      if (sent_message) {
+        printf("BCMP - Received reply to message with seq_num %d\n", header->seq_num);
+        if (xSemaphoreTake(_ctx.messages_list_mutex, pdMS_TO_TICKS(DEFAULT_MESSAGE_TIMEOUT_MS)) == pdPASS) {
+          _message_list_remove_message(sent_message);
+          xSemaphoreGive(_ctx.messages_list_mutex);
+        }
+      } else {
+        // Not one of our messages, so we should ignore it.
+        printf("BCMP - Received message with seq_num %d that we didn't send\n", header->seq_num);
```

Comment:
Will this drop messages that we should forward?

Reply:
I haven't yet tested with 3+ nodes, but it looks like it will block the message! I think we just need to remove the break. I will test this out now!

Reply:
forwarding still works after removing the break.